clients/src/main/java/org/apache/kafka/common/Metric.java

```diff
@@ -27,8 +27,16 @@
     public MetricName metricName();
 
     /**
-     * The value of the metric
+     * The value of the metric as double if the metric is measurable
+     * @throws IllegalStateException if this metric does not have a measurable double value
+     * @deprecated As of 1.0.0, use {@link #metricValue()} instead. This will be removed in a future major release.
      */
+    @Deprecated
     public double value();
 
+    /**
+     * The value of the metric, which may be measurable or a non-measurable gauge
+     */
+    public Object metricValue();
```

Comment:
Would it make sense for `Metric` to be generic?

Reply:
Couldn't think of any value in making this generic. The only place where this method is used is when an Object is returned. Making `Metric` generic would just add a whole load of `Metric<?>` everywhere.

Reply:
It makes me uneasy to use `Object` in a public API, but given that people are already using `Metric` without the type parameter, we can consider whether the cost of updating those usages is worth the benefit later.